From 5afaf602d2dfa5f24b54a9699012439b1b3c205a Mon Sep 17 00:00:00 2001
From: jjm2473 <jjm2473@gmail.com>
Date: Thu, 17 Apr 2025 20:04:38 +0800
Subject: [PATCH] rga2: fix dma_buf api NS not import

```
ERROR: modpost: module rga2 uses symbol dma_buf_vmap from namespace DMA_BUF, but does not import it.
ERROR: modpost: module rga2 uses symbol dma_buf_attach from namespace DMA_BUF, but does not import it.
ERROR: modpost: module rga2 uses symbol dma_buf_map_attachment from namespace DMA_BUF, but does not import it.
ERROR: modpost: module rga2 uses symbol dma_buf_unmap_attachment from namespace DMA_BUF, but does not import it.
ERROR: modpost: module rga2 uses symbol dma_buf_get from namespace DMA_BUF, but does not import it.
ERROR: modpost: module rga2 uses symbol dma_buf_put from namespace DMA_BUF, but does not import it.
ERROR: modpost: module rga2 uses symbol dma_buf_vunmap from namespace DMA_BUF, but does not import it.
ERROR: modpost: module rga2 uses symbol dma_buf_detach from namespace DMA_BUF, but does not import it.
```
---
 drivers/video/rockchip/rga2/rga2_drv.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/video/rockchip/rga2/rga2_drv.c b/drivers/video/rockchip/rga2/rga2_drv.c
index d88bc32..6547932 100644
--- a/drivers/video/rockchip/rga2/rga2_drv.c
+++ b/drivers/video/rockchip/rga2/rga2_drv.c
@@ -2287,3 +2287,6 @@ module_exit(rga2_exit);
 MODULE_AUTHOR("zsq@rock-chips.com");
 MODULE_DESCRIPTION("Driver for rga device");
 MODULE_LICENSE("GPL");
+#if KERNEL_VERSION(5, 16, 0) < LINUX_VERSION_CODE
+MODULE_IMPORT_NS(DMA_BUF);
+#endif
\ No newline at end of file
-- 
2.46.0

